docs/docs-graphql/schema.graphqls

```diff
@@ -825,6 +825,56 @@ The `ContainerID` scalar type represents an identifier for an object of type Con
 """
 scalar ContainerID
 
+"""Reflective module API provided to functions at runtime."""
+type CurrentModule {
+  """A unique identifier for this CurrentModule."""
+  id: CurrentModuleID!
+
+  """The name of the module being executed in"""
+  name: String!
+
+  """
+  The directory containing the module's source code loaded into the engine (plus
+  any generated code that may have been created).
+  """
+  source: Directory!
+
+  """
+  Load a directory from the module's scratch working directory, including any
+  changes that may have been made to it during module function execution.
+  """
+  workdir(
```

Comment:
:bikeshed: This name feels just slightly off since it's sometimes the workdir and sometimes a subdirectory of the workdir (if you pass `path:`).

Having `workdirFile` feels slightly redundant but I see why it's arranged this way; the naive API might be `CurrentModule().Workdir().File("foo")` but that would upload the entire dir, so `WorkdirFile()` helps you avoid the `include` boilerplate. This basically mirrors the `Host().Directory()` and `Host().File()` API, but the `Host()` api avoids the awkward naming because it's just called `Directory()`. :thinking: 

Some options that I'm lukewarm on:

* Add a `Workdir` type that has `Directory()` and `File()` APIs.
* Get rid of `path:` and make you do `Workdir(include, exclude).Directory("foo")`

I don't feel too strongly about this, but figured it's worth a quick non-blocking bikeshed if you agree with any of above. I guess I would prefer the latter, but don't know how painful it is.

Reply:
Yeah agreed, I went in some circles about all this too, but landed here because:
1. Being forced to `include`/`exclude` just to get a file without everything else is very annoying for users
2. Having a completely separate `ModuleWorkdir` type introduces more conundrums since technically that type would be ID-able and thus passable around without having resolved to an actual `Directory`/`File`, but that is hard to imagine since by the time it's passed somewhere, the module workdir may no longer exist (i.e. it's not executing anymore)...

So I'm inclined to stay where it is for now, unless others also disagree.

The only thing I can think of that might help a little bit would be to keep `CurrentModule().Workdir` as is but changed the other api to be `CurrentModule().Workfile` 😄 I'm not sure if that name is too confusing to be worth it though.

Reply:
Sounds good - resolving!